### PR TITLE
Fix default subscriptions, add option for pushshift frontend

### LINKED ISSUE #682: 💡 Feature request: Add option for Pushshift API site
Last year it was requested to switch from reveddit.com to unddit.com as it was considered slow. While that was the case back then, it's the opposite now. Unddit isn't working well recently. Since both of them are interchangeable by just using the other domain it would be great to have an option like `LIBREDDIT_PUSHSHIFT_FRONTEND` where you can set the domain.

### PR DESCRIPTION
**Pull request by sigaloid**
Fixes #682, #689

I combined these two as they're pretty similar - both config related and clearly defined. 

As per #682 perhaps we can consider changing the default to reveddit - is it faster? does it work better? etc.

### CODE REVIEW

**Comment by sigaloid** [comment]
Scratching my head here... :thinking: going to take a look at this later today

File: src/config.rs [hunk 1]
```diff
@@ -52,8 +52,14 @@ pub struct Config {
...
 	pub(crate) default_hide_awards: Option<String>,
 
+	#[serde(rename = "LIBREDDIT_DEFAULT_SUBSCRIPTIONS")]
+	pub(crate) default_subscriptions: Option<String>,
+
 	#[serde(rename = "LIBREDDIT_BANNER")]
 	pub(crate) banner: Option<String>,
+
+	#[serde(rename = "LIBREDDIT_PUSHSHIFT_FRONTEND")]
+	pub(crate) pushshift: String,
```
Comment (r7l) [inline src/config.rs:62]: @sigaloid Should this be the cause for the failed test? Looks like as if this should be `Option<String>`?

Comment (sigaloid) [reply src/config.rs:62]: It needs to be String, as there must be a frontend specified. In any case I don't think that could really make it fail, because it's not affecting the existing test cases... No new tests were added. :/

Comment (r7l) [reply src/config.rs:62]: Ok, i see. I was just trying to have a look at the failed test case but i have to admit that i never worked with Rust before. Just saw that the `get_setting_from_config` has `-> Option<String>` and i would expect that function to always return `Option<String>` which is no longer the case due to `LIBREDDIT_PUSHSHIFT_FRONTEND` being different.

If i may ask a question: How is `LIBREDDIT_BANNER` different from `LIBREDDIT_PUSHSHIFT_FRONTEND`? It is even [defined to be string in the README](https://github.com/libreddit/libreddit#instance-settings).

Comment (sigaloid) [reply src/config.rs:62]: Banner is optional - if there's none set, it won't be displayed. It's listed as string, because not everyone reading the readme would know what `Option<String>` means

Pushshift frontend needs to be defined - it's essentially defined by default with a String rather than an Option<String> to let it fit better in the template

Comment (r7l) [reply src/config.rs:62]: @sigaloid Looks like i can't help here much even if i would love to. But beside Libreddit, i've never had a look at Rust code. Would love to have this in Libreddit but it seems stuck for now. Anyways, thanks allot for explaining and your work on this PR and Libreddit.

Comment (r7l) [reply src/config.rs:62]: @sigaloid Would it be an option to split this PR and try one for the default subscriptions only? It would be great to have this feature back in.

**Comment by Tokarak** [comment]
Found the error using this debug code!
```rust
let config_to_write = r#"LIBREDDIT_DEFAULT_COMMENT_SORT = "best""#;
	write("libreddit.toml", config_to_write).unwrap();
	dbg!(toml::from_str::<Config>(&read_to_string("libreddit.toml").unwrap()));
```
Because `LIBREDDIT_PUSHSHIFT_FRONTEND` is non-optional, it is required during deserialisation. 
It's best to follow standard project practice: use an optional type (the user may or may not set this config), and use `unwrap_or` to fallback into a default value.

It may be also be a good idea to add a test for deserialising from an empty string to avoid this error in the future.
